compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp

```diff
@@ -42,6 +42,11 @@ static llvm::cl::opt<bool> clFailOnOutOfBoundsStackAllocation(
                    "be solved"),
     llvm::cl::init(true));
 
+static llvm::cl::opt<bool> clFailOnLargeVector(
+    "iree-llvmcpu-fail-on-large-vector",
+    llvm::cl::desc("fail if there are operations with large vectors"),
+    llvm::cl::init(true));
+
```

Comment:
Btw, is keeping this pass behind a flag your long-term intent? That would make sense to me, but then one might be tempted to do the same for other passes 😅

Reply:
Yes, I want to keep the flag for long-term. What do you mean by "for other passes"? I think we are enabling the checks for large vectors and stack allocations. Is there something that I'm missing?

Reply:
This was just a generic design question - how do we decide which passes should be hidden behind a flag?

Reply:
I tried to figure a generic answer, but I found that it's case by case.

IMO, there should be a fine-grain flag to control fast-math-like optimization. Unfortunately it is not done yet, so we have several flags here, e.g., `iree-llvmcpu-reassociate-fp-reductions`. IIUC, many ML frameworks just enable it at LLVM level, and pay the costs for accuracy. So people do not care much about it.

For the new option I added and `clFailOnOutOfBoundsStackAllocation`, they are generally good to be enabled by default. Because something is very wrong when it's happening. We want to error out for better user experience and development experience. This kind of flags is useful for developers because they can bypass the check quickly to prefetch further issues. Other users could use them in risks, that's their decision.

For other flags, they are mostly for experimental path, which allows developers smoothly land the support of the new path. This kind of flags need to be killed at some moments.